compiler/rustc_ast_lowering/src/lib.rs

```diff
@@ -977,14 +959,31 @@ impl<'a, 'hir> LoweringContext<'a, 'hir> {
                         span: DUMMY_SP,
                     }
                 };
-                MacArgs::Eq(eq_span, MacArgsEq::Hir(lit))
+                AttrArgs::Eq(eq_span, AttrArgsEq::Hir(lit))
             }
-            MacArgs::Eq(_, MacArgsEq::Hir(ref lit)) => {
+            AttrArgs::Eq(_, AttrArgsEq::Hir(ref lit)) => {
                 unreachable!("in literal form when lowering mac args eq: {:?}", lit)
             }
         }
     }
 
+    fn lower_delim_args(&self, args: &DelimArgs) -> DelimArgs {
+        // This is either a non-key-value attribute, or a `macro_rules!` body. We either do not
+        // have any nonterminals present (in the case of an attribute), or have tokens available
+        // for all nonterminals in the case of a nested `macro_rules`: e.g:
+        // ```rust
+        // macro_rules! outer {
+        //     ($e:expr) => {
+        //         macro_rules! inner {
+        //             () => { $e }
+        //         }
+        //     }
+        // }
+        // ```
+        // In both cases, we don't want to synthesize any tokens
```

Comment:
This comment is largely outdated, we no longer need to synthesize fake tokens almost anywhere, and no longer have methods for synthesizing them here in particular.